ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/OngoingAuthorizationObserver.scala

```diff
@@ -3,14 +3,78 @@
 
 package com.daml.ledger.api.auth
 
+import java.time.Instant
+
+import akka.actor.{Cancellable, Scheduler}
+import com.daml.error.DamlContextualizedErrorLogger
+import com.daml.error.definitions.LedgerApiErrors
+import com.daml.ledger.api.auth.interceptor.AuthorizationInterceptor
+import com.daml.ledger.participant.state.index.v2.UserManagementStore
+import com.daml.lf.data.Ref
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.platform.server.api.validation.ErrorFactories
+import io.grpc.StatusRuntimeException
 import io.grpc.stub.ServerCallStreamObserver
 
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+
+/** @param userRightsCheckIntervalInSeconds - determines the interval at which to check whether user rights state has changed.
+  *                                         Also, double of this value serves as timeout value for subsequent user rights state checks.
+  */
 private[auth] final class OngoingAuthorizationObserver[A](
     observer: ServerCallStreamObserver[A],
-    claims: ClaimSet.Claims,
-    authorized: ClaimSet.Claims => Either[AuthorizationError, Unit],
-    throwOnFailure: AuthorizationError => Throwable,
-) extends ServerCallStreamObserver[A] {
+    originalClaims: ClaimSet.Claims,
+    nowF: () => Instant,
```

Comment:
I see how this could become useful, but AFAICT it is not set to a "fake" clock in any of the tests, is it?

Reply:
It would be useful in a unit test for `OngoingAuthorizationObserver` but such test is missing from this PR